.github/workflows/build-domain.yaml

```diff
@@ -126,30 +127,43 @@ jobs:
           overrides="${service}/chart-overrides.yaml"
           name=$(basename $(yq '.name' "$overrides"))
 
-          version=$(yq '.version' "$overrides")
-          version=$(sh ./.github/tooling/semver.sh "$version" "$PR")
+          tag=$(yq '.version' "$overrides")
+          tag=$(sh ./.github/tooling/semver.sh "$tag" "$PR")
 
-          echo "service=${service}" >> $GITHUB_OUTPUT
-          echo "project=${PROJECT}" >> $GITHUB_OUTPUT
-          echo "docker=${docker}" >> $GITHUB_OUTPUT
-          echo "domain=${domain}" >> $GITHUB_OUTPUT
-          echo "version=${version}" >> $GITHUB_OUTPUT
-          echo "name=${name}" >> $GITHUB_OUTPUT
+          dotnet=${{ matrix.build.dotnetVersion }}
+
+          {
+            echo "service=${service}"
+            echo "project=${PROJECT}"
+            echo "docker=${docker}"
+            echo "domain=${domain}"
+            echo "tag=${tag}"
+            echo "name=${name}"
+            echo "dotnet=${dotnet::${#dotnet}-2}"
+          } >> $GITHUB_OUTPUT
+
+      - name: Pin dotnet version
+        shell: bash
+        working-directory: ${{ steps.resolver.outputs.service }}
+        run: echo '{"sdk":{"version":"${{ inputs.dotnet-version }}"}}' > global.json
```

Comment:
I do not understand why it is needed to generate global.json files at all? Why not just utilize how the dotnet tooling works where it finds the nearest global.json?

Reply:
There needs to be a global.json file "locally" for it to be copied during docker builds. It is only needed if we want to ensure those are built using the specified version.